test/exercise/arrays/solution.rb

```diff
@@ -2,11 +2,53 @@ module Exercise
   module Arrays
     class << self
       def replace(array)
+        ar_len = array.size - 1
+        max = 0
+        while ar_len.positive? do
+          if array[ar_len] > max
+            max = array[ar_len]
+          end
+          ar_len -= 1
+        end
+        while ar_len < array.size do
+          if array[ar_len].positive? && array[ar_len] < max
+            array[ar_len] = max
+          end
+          ar_len += 1
+        end
```

Comment:
it's better to use map/each/reduce/filter e.t.c.
but complexity is OK, no need to change, just FYI